### PR TITLE
workunit: request branch when cloning

### PR DESCRIPTION
**Pull request by dmick**
This is mostly @tchaikov's, from https://github.com/ceph/ceph/pull/14246, which had a few small errors.  It's DNM because I need to test.

### CODE REVIEW

File: qa/tasks/workunit.py [hunk 1]
```diff
@@ -23,14 +23,16 @@ def __init__(self, refspec):
     def __str__(self):
         return self.refspec
 
-    def _clone(self, git_url, clonedir, opts=[]):
+    def _clone(self, git_url, clonedir, opts=None):
+        if opts is None:
+            opts = ''
```
Comment (dmick) [inline qa/tasks/workunit.py]: this needs to be []

**Comment by dmick** [comment]
Tested with teuthology and 1) no ref, 2) 'sha1:', and 3) 'branch:'.  

The final commit also includes a few extra blank lines to appease flake8.

**Comment by tchaikov** [comment]
```

The following tests FAILED:
	  5 - cephtool-test-mon.sh (Timeout)
```

not relevant. i am looking at it.

**Comment by tchaikov** [comment]
see #14266 for the fix.

**Pull request merged**